doc/guide/Guide_continuous_integration_GitHub_actions.ipynb

```diff
@@ -0,0 +1,196 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Continuous Integration and GitHub Actions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Automated Tests\n",
+    "\n",
+    "On Jenkins tests are executed and analyzed automatically, in an unbiased environment. The results are stored and can be compared with previous test runs.\\\n",
+    "Jenkins has a GUI for monitoring individual tests, full test runs and test result trends.\\\n",
+    "Developers are requested to watch it. At first when they push commits to the code repository, but also later on, when other changes in data or sources may make it necessary to review and refactor code that once passed all tests.\n",
+    "The CLIMADA Jenkins server used for continuous integration is at [(https://ied-wcr-jenkins.ethz.ch) ](https://ied-wcr-jenkins.ethz.ch).\n",
+    "\n",
+    "### Developer guidelines:\n",
+    "\n",
+    "- All tests must pass before submitting a pull request.\n",
+    "- Integration tests don't run on feature branches in Jenkins, therefore developers are requested to run them locally.\n",
+    "- After a pull request was accepted and the changes are merged to the develop branch, integration tests may still fail there and have to be addressed."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test Coverage\n",
+    "\n",
+    "Jenkins also has an interface for exploring code coverage analysis result.\\\n",
```

Comment:
First, test coverage is not an issue particular to automated tests. I think general information on coverage should go into the Testing guide. I am also not sure if the example is really helpful. I think it should demonstrate testing code where some important case was missed.

Secondly, executing `make unit_test` and `make integ_tests` provides *local* coverage reports as HTML pages at `coverage/index.html`.

Thirdly, there should be instructions how to inspect coverage reports on Jenkins. E.g., link (for develop): https://ied-wcr-jenkins.ethz.ch/job/climada_branches/job/develop/1089/cobertura/ and image:
<img width="2909" alt="Screenshot 2023-12-06 at 11 26 14" src="https://github.com/CLIMADA-project/climada_python/assets/34276446/0e25ba7b-2a3f-4f4c-917a-507d63f7318f">

Reply:
To the first and second point, yes I can move that to the Testing Guide and make that addition.

To the third, I think it could be helpful to have a look at this together, since I don't have much experience with it yet.

Reply:
@peanutfun reminder to handle this